File: backend/src/main/java/com/freelancing/service/MessageService.java
package com.freelancing.service;

import com.freelancing.dto.request.CreateThreadRequest;
import com.freelancing.dto.request.SendMessageRequest;
import com.freelancing.dto.response.MessageResponse;
import com.freelancing.dto.response.MessageThreadResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface MessageService {
    Page<MessageThreadResponse> getThreads(Long userId, Pageable pageable);
    MessageThreadResponse getThreadById(Long threadId);
    MessageThreadResponse createThread(Long userId, CreateThreadRequest request);
    Page<MessageResponse> getMessages(Long threadId, Pageable pageable);
    MessageResponse sendMessage(Long threadId, Long senderId, SendMessageRequest request);
    void deleteMessage(Long messageId);
    List<MessageThreadResponse> getThreadsByContract(Long contractId);
}

File: frontend/src/api/messageService.js
import api from './axios';

export const messageService = {
  getThreads: () => api.get('/messages/threads'),
  createThread: (data) => api.post('/messages/threads', data),
  getMessages: (threadId, page = 0, size = 20) => api.get(`/messages/threads/${threadId}/messages?page=${page}&size=${size}`),
  sendMessage: (threadId, data) => api.post(`/messages/threads/${threadId}/messages`, data),
  deleteMessage: (id) => api.delete(`/messages/${id}`),
  getThreadByContract: (contractId) => api.get(`/messages/contract/${contractId}`),
};
